pkg/blackboxexporter/blackboxexporter.go

```diff
@@ -0,0 +1,243 @@
+package blackboxexporter
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/const/blackbox"
+	"github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	//api's used
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	//local packages
+)
+
+type BlackboxExporter struct {
+	Client client.Client
+	Log    logr.Logger
+	Ctx    context.Context
+}
+
+func NewBlackboxExporter(client client.Client, log logr.Logger, ctx context.Context) *BlackboxExporter {
+	return &BlackboxExporter{client, log, ctx}
+}
+
+func (b *BlackboxExporter) ShouldDeleteBlackBoxExporterResources() (blackbox.ShouldDeleteBlackBoxExporter, error) {
+
+	routeMonitors := &v1alpha1.RouteMonitorList{}
+	if err := b.Client.List(b.Ctx, routeMonitors); err != nil {
+		return blackbox.KeepBlackBoxExporter, err
+	}
+
+	clusterUrlMonitors := &v1alpha1.ClusterUrlMonitorList{}
+	if err := b.Client.List(b.Ctx, clusterUrlMonitors); err != nil {
+		return blackbox.KeepBlackBoxExporter, err
+	}
+
+	if len(clusterUrlMonitors.Items)+len(routeMonitors.Items) > 1 {
```

Comment:
this means that if we passed that if we have three scenarios:
1. there is 1 routeMonitor and no clusterUrlMonitor
2. there is 1 clusterUrlMonitor and no routeMonitor
3. there are none of both

wouldn't that be clearer to write in if statements then range on the items?

Reply:
You don't think it's clear that we do not want to remove if both lists together contain more than 1 item?

Reply:
what I mean is that the steps after that were not changed..

testing if ALL the resources are not deleting makes it seem like there is more than one.

Reply:
refactored that function.